File: src/main/java/fm/comunas/ddd/oe/model/ObjUserRepository.java

package fm.comunas.ddd.oe.model;

import fm.comunas.ddd.obj.model.ObjRepository;
import fm.comunas.ddd.oe.model.db.tables.records.ObjUserVRecord;

import java.util.Optional;

public interface ObjUserRepository extends ObjRepository<ObjUser, ObjUserVRecord> {

	/**
	 * Lookup user with given id (in global session)
	 */
	Optional<ObjUser> get(Integer id);

	/**
	 * Lookup User with email
	 */
	Optional<ObjUser> getByEmail(String email);

}

File: fm-dddrive/src/test/java/dddrive/domain/oe/model/ObjUserRepository.kt
package dddrive.domain.oe.model

import dddrive.app.obj.model.ObjRepository
import io.dddrive.oe.model.ObjUser
import java.util.*

interface ObjUserRepository : ObjRepository<ObjUser> {

	fun getByEmail(email: String): Optional<ObjUser>

	companion object {

		const val KERNEL_USER_EMAIL: String = "k@dddrive.io"
	}
}
